lib/index.js

```diff
@@ -196,13 +197,31 @@ Socket.prototype._stopPollReceive = function () {
  */
 
 Socket.prototype.bind = function (addr) {
-    return this._protect(nn.Bind(this.binding, addr));
+  var eid = nn.Bind( this.binding, addr );
+  if(eid < 0){
+    return this.emit('error', new Error(nn.Err() +': '+this.type+' bind@' + addr));
+  }
+  this.how[addr] = [eid,'bind'];
+  return this.how; //socket's active endpoints
 }
 
 Socket.prototype.connect = function (addr) {
-    return this._protect(nn.Connect(this.binding, addr));
+  var eid = nn.Connect( this.binding, addr )
+  if(eid < 0){
+    return this.emit('error', new Error(nn.Err() +': '+this.type+' connect@' + addr));
+  }
+  this.how[addr] = [eid,'connect'];
+  return this.how; //socket's active endpoints
 }
 
+Socket.prototype.shutdown = function (addr) {
+  var confirm = this.how[addr][1] + ' endpoint '+ addr + ' shutdown';
+  if(nn.Shutdown(this.binding, this.how[addr][0]) < 0){
+    return this.emit('error', new Error(nn.Err()+': '+this.type+' shutdown@'+addr));
+  }
+  delete this.how[addr]; return confirm;
+};
```

Comment:
... then here it would be:
```js
if(Object.keys(this.binds).conatains(addr)) {
 ...
  delete this.binds[addr];
} else if (Object.keys(this.connects).contains(addr)) {
 ...
  delete this.connects[addr];
}
```